classes/Translate.php

```diff
@@ -129,7 +129,7 @@ public static function getModuleTranslation(
         // $translations_merged is a cache of wether a specific module's translations have already been added to $_MODULES
         static $translationsMerged = [];
 
-        $name = $module instanceof Module ? $module->name : $module;
+        $name = $module instanceof ModuleCore ? $module->name : $module;
```

Comment:
This might be a change in the behavior.

If I have correctly understood the legacy system,
-  ModuleCore is the original class
- there is a stub called Module that extends ModuleCore in default case
- but if an override is done against Module, then Module extends the override, not ModuleCore

So if the above is good
- the code on `develop` will work with overrides to Module that do not extend ModuleCore
- the changes you submitted do not work with overrides to Module that do not extend ModuleCore

That being said I think overrides are supposed to extend the overriden class.